### ISSUE TITLE
Confusing categorisation of Vanilla reference

### ISSUE THREAD

**Issue by matthewpaulthomas** [ubuntudesign/docs.vanillaframework.io#108]
[The current reference](https://docs.vanillaframework.io/) is arranged in five categories:
- Settings: Assets, Breakpoints, Color, Font, Layout, Spacing
- Base: Code, Forms, Reset, Table, Typography (headings, ordered/unordered/description lists, blockquotes and citations, and some inline text formatting)
- Patterns: Breadcrumbs, Buttons, Card, Code numbered, Code snippet, Footer, Grid, Inline images, Links, Lists (simple, icon/dividers/both, middot-separated, stepped), Matrix, Navigation (header), Notification, Pull quote, Strip (container for a grid row)
- Utilities: Align (centered/left/right), Clearfix, Embedded media, Equal height, Floats, Margin collapse, Padding collapse, Off screen, Vertically center, Visibility
- Themes

I’m told that this structure is based on “[Inverted Triangle CSS](https://www.xfive.co/blog/itcss-scalable-maintainable-css-architecture/)”, in which rules are organized in seven layers, from most generic to most specific: “Settings”, “Tools”, “Generic”, “Elements”, “Objects”, “Components”, and “Trumps”/“Utilities”.

Even if you are familiar with ITCSS, Vanilla makes this structure mostly unrecognisable by combining “Generic” and “Elements” into “Base”, combining “Objects” and “Components” into “Patterns”, and adding “Themes” at the end.

But regardless of whether you are familiar with ITCSS, this structure is more relevant to someone _contributing to_ Vanilla than someone actually _using_ Vanilla. From the perspective of someone wanting to use Vanilla patterns, the navigation seems completely disorganized. For example:
- The eight ways to format a (non-definition) list are spread over two pages: two in the middle of “Base” > “Typography”, and six in “Patterns” > “List”.
- The five ways to format code are spread over _three_ pages: “Base” > “Code”, “Patterns” > “Code numbered”, and “Patterns” > “Code snippet”.
- There are two styles for block quotes, one near the end of “Base” > “Typography”, and one in “Patterns” > “Pull quote”.
- One kind of form control is described in “Patterns” > “Buttons”, some others are described in “Base” > “Forms”, and the rest are mentioned in “Base” > “Forms” but not illustrated.

There are also a couple of cases of confusing terminology:
- “Patterns” > “Strip” is not usefully defined, but turns out to be a small part of the “Patterns” > “Grid” component.
- “Patterns” > “Navigation” could potentially include menus, breadcrumbs, tabs, and footers, but in this case refers only to headers.

The reference should be recategorized to suit people who are looking to use Vanilla to style a document or app.

**Comment by matthewpaulthomas** [comment]
Card sorting:
![card-sorting](https://user-images.githubusercontent.com/19801137/28367691-cd270c3e-6c89-11e7-9b59-123d8aa2ebb7.png)

**Comment by matthewpaulthomas** [comment]
Proposed categorization:

# Basics
- Font (including Cyrillic+Greek)
- Color
- Spacing
- Reset (if it needs mentioning at all)

# Global layout
- Grid (including current “Layout”, which is grid dimensions)
- Strip
- Breakpoints

# One-off layout
- Alignment (horizontal and vertical)
- Padding/margin collapsing
- Equal height
- Off-screen elements
- Floats and clearing
- Visibility ([?](https://github.com/ubuntudesign/docs.vanillaframework.io/issues/107))

# Navigation
- Header
- Footer
- Breadcrumbs
- (Tabs?)

# Page structure
- Heading (including heading classes)
- Notification
- Table
- Card
- Inline images
- Matrix
- List (unordered, ordered, definition, and all others)

# Text
- Inline formatting (small, strong, superscript…)
- Code (including snippet and numbered code)
- Quotes (block quotes and pull quotes)

# Interactive elements
- Link
- Button
- Text field
- File picker
- Color picker
- Date picker
- …

# Media
- Embedded media
- Assets

# Themes

**Comment by kwm14** [comment]
Thanks @matthewpaulthomas 

Reading through your proposal you make valid points, but we will remove 'Themes' as this is no longer valid. I'll create an issue for implementation by development to update categorisation in Vanilla.

+1

**Comment by matthewpaulthomas** [comment]
@kwm14 Excellent. Themes were the big flaw in my proposal, because each theme can/could include patterns that would fit into some of the general categories I identified. Let me know if I can help in categorizing the theme patterns.

(I’m sorry I reported this issue on the wrong repo: I assumed that docs.vanillaframework.io covered the docs itself, and [the README](https://github.com/ubuntudesign/docs.vanillaframework.io/blob/master/README.md) misleadingly refers to “the content of this project” without mentioning that that content is in a different repo entirely.)
